src/test/java/org/springframework/integration/aws/config/xml/SnsOutboundChannelAdapterParserTests.java

```diff
@@ -79,6 +80,8 @@ void testSnsOutboundChannelAdapterDefaultParser() {
 		assertThat(TestUtils.getPropertyValue(this.defaultAdapterHandler, "amazonSns")).isSameAs(this.amazonSns);
 		assertThat(TestUtils.getPropertyValue(this.defaultAdapterHandler, "evaluationContext")).isNotNull();
 		assertThat(TestUtils.getPropertyValue(this.defaultAdapterHandler, "topicArnExpression")).isNull();
+		assertThat(TestUtils.getPropertyValue(this.defaultAdapterHandler, "messageGroupIdExpression")).isNull();
+		assertThat(TestUtils.getPropertyValue(this.defaultAdapterHandler, "messageDeduplicationIdExpression")).isNull();
```

Comment:
`isNull()`?
How about to have a test when it is not null?
Both of them, of course.

I believe for that purpose we might need to have additional channel adapter config and other unit test method in this class.
That `id="defaultAdapter"` is really to check what default values are populated for the properties.
Yes, in case of default these your new props are null.
Now let's see if we can have a separate `<int-aws:sns-outbound-channel-adapter>` where these props are not null.

Reply:
I am not familiar with how to construct these tests and have copied the existing model, which tests _only_ for this case (that is, there are not any tests for the existing XML attributes).

Reply:
OK. Fair enough. I will take a look into that locally when you address the rest of my review and this one will be ready for merging.